lib/spack/spack/main.py

```diff
@@ -709,7 +712,7 @@ def main(argv=None):
     if not args.no_env:
         env = ev.find_environment(args)
         if env:
-            ev.activate(env, args.use_env_repo)
+            ev.activate(env, args.use_env_repo, not args.no_env_view)
```

Comment:
I don't think we need an argument here, this should never activate the view with the environment. Commands other than `spack env activate` and `spack env deactivate` aren't equipped to handle the relevant environment changes, and those commands activate/deactivate the environment later (in the SpackCommand()()) call.

Reply:
If we always pass `add_view=False`, then detecting this issue works well when trying to activate an environment with a missing package/namespace, but it won't detect when this happens and you're already in the environment. Maybe we should have the `spack env status` command also validate that all packages still exist?